core/src/main/scala/libretto/scaletto/ScalettoLib.scala

```diff
@@ -281,27 +278,26 @@ class ScalettoLib[
   : (A |*| B) -⚬ ((A |*| B) |+| (B |*| A)) =
     lteqBy(aKey, bKey).>.right(swap)
 
-  implicit def comparableVal[A](using A: Ordering[A]): Comparable[Val[A], Val[A]] =
-    new Comparable[Val[A], Val[A]] {
-      import coreLib.Compared._
+  given [A : Ordering]: Comparable[Val[A], Val[A]] with {
+    import coreLib.given, Compared._, Either as ⊻
 
-      private val scalaCompare: ((A, A)) => ((A, A) Either ((A, A) Either (A, A))) =
-        { (a1, a2) =>
-          A.compare(a1, a2) match {
-            case i if i < 0 => Left((a1, a2))
-            case 0          => Right(Left((a1, a2)))
-            case i if i > 0 => Right(Right((a1, a2)))
-          }
+    private val scalaCompare: ((A, A)) => ((A, A) ⊻ ((A, A) ⊻ (A, A))) =
```

Comment:
Imho better readable infix `Either`. But not sure whether you like it.

Reply:
👍